src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -454,6 +478,86 @@ class IpexCommunicationService extends AgentService {
       notifRecord.a.r as NotificationRoute
     );
   }
+
+  @OnlineOnly
+  async grantAcdcById(credentialId: string) {
+    const pickedCred = await this.props.signifyClient
+      .credentials()
+      .get(credentialId)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (/500/gi.test(status)) {
+          // TODO: Should change to 404 once KERIA fixed
+          throw new Error(`${Agent.MISSING_DATA_ON_KERIA}: ${credentialId}`);
+        } else {
+          throw error;
+        }
+      });
+
+    if (!pickedCred) {
+      throw new Error(
+        `${IpexCommunicationService.CREDENTIAL_NOT_FOUND} ${credentialId}`
```

Comment:
just throw this inside the `/500` error above. `MISSING_DATA_ON_KERIA` is only used when fetching from the UI normally. So we can remove this if statement and update the error above